allensdk/test/brain_observatory/behavior/behavior_project_cache_data_model/test_behavior_project_cache.py

```diff
@@ -0,0 +1,250 @@
+import pytest
+import pandas as pd
+import logging
+import numpy as np
+import os
+
+
+def safe_df_comparison(expected: pd.DataFrame, obtained: pd.DataFrame):
```

Comment:
My understanding of this function is that it is reimplementing some of `pd.testing.assert_frames_equal` by providing more useful error messages and better handling of null values. I think this function is more general than just behavior_project_cache. Maybe move to `comparison_utils.py`? Also maybe make more general by parameterizing whether columns need to be in an expected order.

Reply:
I've moved it to `allensdk/test_utilities/custom_comparators.py`